### PR TITLE
[Phase 3] MSSQL JSON - test fixture + REST CRUD tests

### LINKED ISSUE #3721: [Phase 3] MSSQL JSON - test fixture + REST CRUD tests

### PR DESCRIPTION
**Pull request by souvikghosh04**
## What this PR does

Adds the database test fixture and REST integration tests for SQL Server 2025 `JSON` columns. It builds on Phase 2 (which taught the engine to treat `JSON` as a normal `string`) and proves the behavior end-to-end through the REST API.

This follows the exact pattern of the recently merged **vector data type** feature (#3677), and is unblocked by CI now running SQL Server 2025 (#3697).

## What's included

- **Test table** — a `profiles` table with an `id` and a `metadata json` column, plus 5 seed rows covering a simple object, an array, a deeply nested object, unicode + emoji, and `NULL`.
- **Entity** — a `Profile` entity (REST + GraphQL enabled) in the MsSql test config and config generator.
- **REST tests** — `MsSqlRestJsonTypesTests`: read (list, by id, null, array, nested, unicode), insert, update (PUT/PATCH), clear-to-null, and delete.

Because DAB treats `JSON` as a string, the tests parse `metadata` and compare it **semantically**, so they're robust to any whitespace / key-order normalization the engine applies.

## No gating needed

Unlike the earlier plan, the table is created **unconditionally** (no server-version guard) — CI runs SQL Server 2025 and the schema already uses the 2025-only `vector` type the same way.

## Delivery plan

| Phase | What | Status |
|------|------|--------|
| 1 | .NET 10 + SqlClient 6.x (prerequisite) | ✅ Merged (#3697/#3656) |
| 2 | JSON type + error mapping (engine) | ✅ Merged (#3691) |
| **3** | **Test fixture + REST CRUD tests (this PR)** | 🚧 In review |
| 3b | GraphQL / OpenAPI / MCP schema-discovery tests | Next |
| 4 | Error/filter edge cases + regression | After |

## Notes
- Requires SQL Server 2025 / Azure SQL (native `json` type).
- No changes to PostgreSQL, MySQL, DwSql, or CosmosDB.
- Related issue: #2768.

### CODE REVIEW

File: src/Service.Tests/SqlTests/RestApiTests/MsSqlRestJsonTypesTests.cs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+            }
+
+            await DeleteProfile(newId);
+        }
+
+        /// <summary>
+        /// PUT /api/Profile/id/1 - Verify a full update replaces the metadata payload, then restore it.
+        /// </summary>
+        [TestMethod]
+        public async Task PutJsonType_Update()
```
Comment (aaronburtle) [inline src/Service.Tests/SqlTests/RestApiTests/MsSqlRestJsonTypesTests.cs:158]: PutJsonType_Update, PatchJsonType_Update, and PatchJsonType_ToNull all mutate shared rows, and then restore with a follow-up call. If any assertions fail before the restore, then the rows are corrupted for any tests that run afterwards. Consider wrapping the mutate/assert/restore in a try-finally for safety.

ie:
try
{
    // mutate + assert
}
finally
{
    // restore original value
}

Comment (aaronburtle) [reply src/Service.Tests/SqlTests/RestApiTests/MsSqlRestJsonTypesTests.cs:158]: We have a similar smell in our current vector tests that we should probably fix too, but not in this PR.

**Review by aaronburtle — APPROVED** [review]
Looks good!

File: src/Service.Tests/SqlTests/RestApiTests/MsSqlRestJsonTypesTests.cs (hunk 1 above)
Comment (souvikghosh04) [reply src/Service.Tests/SqlTests/RestApiTests/MsSqlRestJsonTypesTests.cs:158]: Good catch - fixed in b2ab68f8. Wrapped the mutate/assert in a try and moved the restore into a finally for all three tests (PutJsonType_Update, PatchJsonType_Update, PatchJsonType_ToNull) so the shared row is always reset even if an assertion fails. Left the identical smell in the vector tests alone per your follow-up note.

**Review by Aniruddh25 — APPROVED** [review]
